go/types/intersect_test.go

```diff
@@ -0,0 +1,88 @@
+package types
+
+import (
+	"testing"
+
+	"github.com/attic-labs/testify/assert"
+)
+
+func TestIntersectImpl(t *testing.T) {
```

Comment:
add some tests which cover hit & miss cases for containers with union parameters. e.g. List<A|B|C> & List<C|D>.

Also maybe one "recursive" struct case, like {a: List<B>} & {a: List<A|B>}